python/kfserving/kfserving/storage.py

```diff
@@ -78,6 +82,30 @@ def _download_gcs(uri, temp_dir: str):
                     os.makedirs(local_object_dir, exist_ok=True)
             blob.download_to_filename(os.path.join(temp_dir, subdir_object_key))
 
+    @staticmethod
+    def _download_blob(uri, temp_dir: str):
+        m = re.search(_BLOB_RE, uri)
+        account_name = m.group(1)
+        storage_url = m.group(2)
+        container_name, blob_url = storage_url.split("/", 1)
+
+        block_blob_service = BlockBlobService(account_name=account_name)
+        blobs = block_blob_service.list_blobs(container_name, prefix=blob_url)
+
+        for blob in blobs:
+            if "/" in blob.name:
```

Comment:
seems like the only thing that needs doing inside this block is to create the directory if required? all three calls to get_blob_to_path are identical... and can just be done once at the end?